src/vm/dllimport.cpp

```diff
@@ -7047,7 +7055,6 @@ HINSTANCE NDirect::LoadLibraryModule(NDirectMethodDesc * pMD, LoadLibErrorTracke
     bool libNameIsRelativePath = Path::IsRelative(wszLibName);
     if (hmod == NULL)
     {
-#ifndef FEATURE_CORECLR
         // First checks if the method has DefaultDllImportSearchPathsAttribute. If method has the attribute
```

Comment:
@gkhanna79 The problem is the policy from the host is too coarse-grained (it is a simply a list of directories, after all) for the interop search policies. Either we re-design the host policy based on the interop policy and decide the priority for each sub-policy (which is probably going to be fairly complicated and expensive), or we go with the simple policy which is host overrides the interop policy, which is what this change does. It looks like we haven't really thought out / test the servicing scenario yet (which should be orthogonal to this change), and we should track it using a different issue.